src/main/scala/leon/evaluators/RecursiveEvaluator.scala

```diff
@@ -231,7 +232,47 @@ abstract class RecursiveEvaluator(ctx: LeonContext, prog: Program, maxSteps: Int
 
     case RealMinus(l,r) =>
       e(RealPlus(l, RealUMinus(r)))
-
+      
+    case StringConcat(l, r) =>
+      (e(l), e(r)) match {
+        case (StringLiteral(i1), StringLiteral(i2)) => StringLiteral(i1 + i2)
+        case (le,re) => throw EvalError(typeErrorMsg(le, StringType))
+      }
+    case StringLength(a) => e(a) match {
+      case StringLiteral(a) => InfiniteIntegerLiteral(a.length)
+      case res => throw EvalError(typeErrorMsg(res, IntegerType))
+    }
+    case SubString(a, start, end) => (e(a), e(start), e(end)) match {
+      case (StringLiteral(a), InfiniteIntegerLiteral(b), InfiniteIntegerLiteral(c))  =>
+        StringLiteral(a.substring(b.toInt, c.toInt))
+      case res => throw EvalError(typeErrorMsg(res._1, StringType))
+    }
+    case Int32ToString(a) => e(a) match {
+      case IntLiteral(i) => StringLiteral(i.toString)
+      case res =>  throw EvalError(typeErrorMsg(res, Int32Type))
+    }
+    case CharToString(a) => 
+      e(a) match {
+        case CharLiteral(i) => StringLiteral(i.toString)
+        case res =>  throw EvalError(typeErrorMsg(res, CharType))
+      }
+    case IntegerToString(a) => e(a) match {
+        case InfiniteIntegerLiteral(i) => StringLiteral(i.toString)
+        case res =>  throw EvalError(typeErrorMsg(res, IntegerType))
+      }
+    case BooleanToString(a) => e(a) match {
+        case BooleanLiteral(i) => StringLiteral(i.toString)
+        case res =>  throw EvalError(typeErrorMsg(res, BooleanType))
+      }
+    case RealToString(a) => e(a) match {
+        case FractionalLiteral(n, d) => StringLiteral(n.toString + "/" + d.toString)
+        case res =>  throw EvalError(typeErrorMsg(res, RealType))
+      }
+    case StringEscape(a) => e(a) match {
```

Comment:
For no reason, I could not include a call to `org.apache.commons.lang3.StringEscapeUtils` in the library, which would have allowed me to only use a static library call.
It complained with `apache` is not a member of `org` at run time.
This is why I reverted to a custom `StringEscape` tree.

Any workaround? Me too I prefer to have library calls rather than new trees.